### PR TITLE
feat: SafepalWallet hide "Add to TrustWallet"

### PR DESCRIPTION
**Pull request by Woody810**

### CODE REVIEW

File: src/utils/wallet.ts [hunk 1]
```diff
@@ -105,5 +105,6 @@ export const canRegisterToken = () =>
   typeof window !== 'undefined' &&
   (window?.ethereum?.isMetaMask ||
     window?.ethereum?.isTrust ||
+    window?.ethereum?.isSafePal ||
```
Comment (chef-jojo) [inline src/utils/wallet.ts:108]: It seems safapal doesn't support watch token, let's return false if safepal

File: src/utils/wallet.ts [hunk 2]
```diff
@@ -104,6 +104,8 @@ export const registerToken = async (
 export const canRegisterToken = () =>
   typeof window !== 'undefined' &&
   (window?.ethereum?.isMetaMask ||
-    window?.ethereum?.isTrust ||
+    (window?.ethereum?.isTrust &&
+      // @ts-ignore
```
[the remark this answers is not in the document]
Comment (Woody810) [reply src/utils/wallet.ts:108]: Yes, if not added, there will show " type 'Ethereum' has no attribute 'isSafePal' "

File: packages/uikit/src/react-app-env.d.ts [hunk 3]
```diff
@@ -4,6 +4,8 @@ interface Window {
     isOpera?: true;
     isCoinbaseWallet?: true;
     isTrust?: true;
+    // @ts-ignore
```
Comment (chef-jojo) [inline packages/uikit/src/react-app-env.d.ts:7]: don't need in d.ts

**Comment by chef-jojo** [comment]
let's try to fix the type later. I think it's because of wagmi global type Ethereum

File: src/components/AddToWallet/AddToWalletButton.tsx [hunk 4]
```diff
@@ -45,9 +46,14 @@ const getWalletIcon = (marginTextBetweenLogo: string) => {
 }
 
 const getWalletName = () => {
-  if (window?.ethereum?.isTrust) {
+  // @ts-ignore
+  if (window?.ethereum?.isTrust && !window?.ethereum?.isSafePal) {
     return 'Trust Wallet'
   }
+  // @ts-ignore
+  if (window?.ethereum?.isSafePal) {
```
Comment (chef-jojo) [inline src/components/AddToWallet/AddToWalletButton.tsx:54]: maybe just move this before `isTrust`?

Comment (Woody810) [reply src/components/AddToWallet/AddToWalletButton.tsx:50]: done

File: packages/uikit/src/react-app-env.d.ts (hunk 3 above)
Comment (Woody810) [reply packages/uikit/src/react-app-env.d.ts]: done

**Review by chef-jojo — APPROVED** [review]

**Pull request merged**